test/Microsoft.AspNetCore.Mvc.Core.Test/RedirectToActionResultTest.cs

```diff
@@ -83,6 +83,41 @@ public class RedirectToActionResultTest
                 "No route matches the supplied values.");
         }
 
+        [Fact]
+        public async Task RedirectToAction_Execute_WithFragment_PassesCorrectValuesToRedirect()
+        {
+            // Arrange
+            var expectedUrl = "/Home/SampleAction#test";
+            var expectedPermanentFlag = false;
+
+            var httpContext = new Mock<HttpContext>();
+            httpContext
+                .SetupGet(o => o.RequestServices)
+                .Returns(CreateServices().BuildServiceProvider());
+
+            var httpResponse = new Mock<HttpResponse>();
+            httpContext
+                .Setup(o => o.Response)
+                .Returns(httpResponse.Object);
+
+            var actionContext = new ActionContext(httpContext.Object, new RouteData(), new ActionDescriptor());
+
+            var urlHelper = GetMockUrlHelper(expectedUrl);
+            var result = new RedirectToActionResult("SampleAction", "Home", null, false, "test")
+            {
+                UrlHelper = urlHelper,
+            };
+
+            // Act
+            await result.ExecuteResultAsync(actionContext);
+
+            // Assert
+            // Verifying if Redirect was called with the specific Url and parameter flag.
```

Comment:
nit: "the specific Url and parameter flag" -> "the expected URL and permanent parameters"